matlab/util/+io/+fits/writeSimpleFITS.m

```diff
@@ -26,13 +26,30 @@ function writeSimpleFITS(Image, FileName, Args)
     if isempty(Args.Header)
         % create minimal default FITS header
         Args.Header = io.fits.defaultHeader(Args.DataType, size(Image));
-    else
-        % update the BITPIX key val if needed
-        BitPix      = io.fits.dataType2bitpix(Args.DataType);
-        Args.Header = imUtil.headerCell.replaceKey(Args.Header,'BITPIX',{BitPix});
     end
+    % update the BITPIX and BZERO key vals if needed (or, should we keep
+    %  them in some particular case?)
+    [BitPix,bzero]  = io.fits.dataType2bitpix(Args.DataType);
+    Args.Header = imUtil.headerCell.replaceKey(Args.Header,'BITPIX',{BitPix});
+    Args.Header = imUtil.headerCell.replaceKey(Args.Header,'BZERO',{bzero});
+
     HeaderStr = io.fits.generateHeaderBlocks(Args.Header);
 
+    % if the class is unsigned, we must write Image-bzero as signed, and
+    % change the required class.
+    % To do the substraction, we have to upcast
+    switch Args.DataType
+        case 'uint8'
+            Image=int8(int16(Image)-int16(bzero));
+            Args.DataType='int8';
+        case 'uint16'
+            Image=int16(int32(Image)-int32(bzero));
+            Args.DataType='int16';
+        case 'uint32'
+            Image=int32(int64(Image)-int64(bzero));
+            Args.DataType='int16';
```

Comment:
this should be int32, correct

Reply:
corrected in the subsequent commit 5c709e3716, use that instead
Btw, this up and downconverting implies a time penaly, scan't help wit that